### PR TITLE
Feature: Introduce new payload parameters

### PR DESCRIPTION
**Pull request by anouarchattouna**
In [release 0.7.0 of GCR Cleaner tool,](https://github.com/GoogleCloudPlatform/gcr-cleaner/releases/tag/v0.7.0) there was an introduction of new parameters:

> This release introduces a new mechanism for deleting tagged images. You can now specify tag_filter_all or tag_filter_any to control how the filters match against images with multiple tags. The existing behavior of tag_filter is preserved, but will be removed in a future release. Additionally, specifying allow_tagged is no longer necessary as specifying any tag filter will automatically activate deleting tagged images.

Also in this PR:
* Updating examples
* Updating documentation
* Updating pre-commit repos URLs: using https:// instead of git://. See https://github.blog/2021-09-01-improving-git-protocol-security-github/
for more details

### CODE REVIEW

**Review by madhifallah — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -71,23 +71,10 @@ module "gcr_cleaner" {
       project_id     = "yet-another-project-id"
       clean_all      = true
       parameters = {
-        allow_tagged = true
```
Comment (elagree) [inline README.md:74]: Is there any reason to remove allow_tagged ?

File: variables.tf [hunk 2]
```diff
@@ -62,22 +62,26 @@ variable "gcr_repositories" {
...
-      recursive    = If set to true, will recursively search all child repositories (optional(bool))
+    project_id     = Value of the Google project id, if ommited, it will be assigned `google_project_id` variable value (optional(string))
+    storage_region = Location of the storage bucket (optional(string))
+    repositories = Docker image repositories to clean (optional(list(object({
+      name           = Name of the repository (string)
+      grace          = Relative duration in which to ignore references. This value is specified as a time duration value like "5s" or "3h". If set, refs newer than the duration will not be deleted. If unspecified, the default is no grace period (all untagged image refs are deleted) (optional(string))
+      allow_tagged   = (Deprecated) This option is deprecated and has no effect. By default, GCR Cleaner will not delete tagged images. To delete tagged images, specify `tag_filter_any` or `tag_filter_all` (optional(bool))
```
Comment (elagree) [inline variables.tf:70]: auto answer seems yes^^

**Review by elagree — APPROVED** [review]
Thanks

File: variables.tf [hunk 3]
```diff
@@ -87,19 +91,23 @@ EOF
...
     repositories = optional(list(object({
-      name         = string
-      grace        = optional(string)
-      allow_tagged = optional(bool)
-      keep         = optional(string)
-      tag_filter   = optional(string)
-      recursive    = optional(bool)
+      name           = string
+      grace          = optional(string)
+      allow_tagged   = optional(bool)
```
Comment (tbeaugrand) [inline variables.tf:96]: remove this, no ?

File: variables.tf [hunk 4]
```diff
@@ -87,19 +91,23 @@ EOF
...
+      recursive      = optional(bool)
     })))
     clean_all = optional(bool)
     parameters = optional(object({
-      grace        = optional(string)
-      allow_tagged = optional(bool)
-      keep         = optional(string)
-      tag_filter   = optional(string)
+      grace          = optional(string)
+      allow_tagged   = optional(bool)
```
Comment (tbeaugrand) [inline variables.tf:106]: same

**Review by tbeaugrand — APPROVED** [review]

**Pull request merged**